Sources/Internal/BridgeTransport.swift

```diff
@@ -6,10 +6,13 @@ import Foundation
 
 protocol Transport {
     func send(to url: WCURL, text: String)
-    func listen(on url: WCURL,
-                onConnect: @escaping ((WCURL) -> Void),
-                onDisconnect: @escaping ((WCURL, Error?) -> Void),
-                onTextReceive: @escaping (String, WCURL) -> Void)
+    func listen(
+        on url: WCURL,
+        onConnect: @escaping ((WCURL) -> Void),
+        onDisconnect: @escaping ((WCURL, Error?) -> Void),
+        onTextReceive: @escaping (String, WCURL) -> Void,
+        onDeserializationError: @escaping ((WCURL) -> Void)
```

Comment:
I think instead of a specific error case, it would be better to have an error closure i.e. `onError: @escaping ((WCURL, Error?) -> Void)`. That way, we can use it to add other errors. Since, we will drop supporting v1 at some point, I don't know it is critical. I am leaving the decision to you.

Reply:
@sakkaras I thought the same but I think it's not possible to track and have a meaningful data about all the errors that we receive from the SDK. So, I decided to limit the error with the scope of what've seen and focus on that part. What do you think?

Reply:
I think I don't need to answer that since you had already made the change. Thanks!

Reply:
Yes, I added a generic error structure but only called the error response when there is a deserialization error for now.